src/utils/bitfield.rs

```diff
@@ -0,0 +1,231 @@
+use core::ops::{Bound, Range, RangeBounds};
+
+/// Trait for manipulating subsets of integers.
+pub trait BitField {
+    const TYPE_SIZE: usize;
+
+    /// Return a boolean representing the state of the idx'th bit.
+    ///
+    /// # Arguments
+    ///
+    /// * `idx` - The index of the bit, 0 being the rightmost one.
+    ///
+    /// # Panics
+    ///
+    /// This method will panics if the given index is out of range.
+    fn get_bit(self, idx: usize) -> bool;
+
+    /// Return a value representing the state of a range of bits.
+    ///
+    /// # Arguments
+    ///
+    /// * `range` - The range of bits, starting from 0 for the rightmost one.
+    ///
+    /// # Panics
+    ///
+    /// This method will panics if the given range is invalid.
+    fn get_bits<T: RangeBounds<usize>>(self, range: T) -> Self;
+
+    /// Set the idx'th bit.
+    ///
+    /// # Arguments
+    ///
+    /// * `idx` - The index of the bit, 0 being the rightmost one.
+    /// * `value` - A boolean representing the desired state of the bit.
+    ///
+    /// # Panics
+    ///
+    /// This method will panics if the given index is out of range.
+    fn set_bit(self, idx: usize, value: bool) -> Self;
```

Comment:
We may need to put those functions as const functions so they may be executed at compile time.

Reply:
After investigating this further, it is not feasible. Range types implement the `Drop` trait and therefore cannot be used in const functions.

Reply:
Thank you very much for confirming